### ISSUE TITLE
Implement Python 3.11 support in 2U Owned repos

### ISSUE THREAD

**Issue by UsamaSadiq** [edx/edx-arch-experiments#553]
## Disclaimer 
- At first, `Python 3.12` was the target version, but after identifying major changes needed for `Python 3.12` support, it was decided to release the `redwood` version with `Python 3.11` support instead. The following pending PRs have been updated instead to add `Python 3.11` support.

## Description
- Upgrade following repositories to add support for `Python 3.11` testing and drop testing for `Django 3.2`
- Once the major services have been upgraded, upgrade the packages core python version to 3.11 and drop support for `Python<3.11` in a follow up issue. 

## Instructions
Complete the following checklist where applicable to mark the PR as Done

### Check list  
- Python 3.11 has been added to `tox` and `github actions CI`
- Django 3.2 has been removed from `tox` and `github actions CI`
- The tests shards with `Python 3.11` are passing
- Package version has been bumped after updating CHANGELOG entry
- A new GitHub release has been created after merging the Python support PR

```[tasklist]
### Repositories
- [x] edx-analytics-dashboard [IDA]
- [ ] xblock-skill-tagging [No write permissions to arbi-bom team]
- [x] taxonomy-connector [Usama] [In author team review]
- [x] edx-analytics-data-api [IDA]
- [x] registrar [IDA] [Usama] [Blocked on owning team to upgrade celery]
- [x] edx-platform [IDA] [Feanil - Axim]
- [x] credentials [IDA] [Awais - wg-maintenance]
- [x] license-manager [IDA] [Umar] [Support Added]
- [x] course-discovery [IDA] [Soban]
- [x] enterprise-access [IDA]
- [x] enterprise-catalog [IDA] [Usama] [3.12 support added]
- [x] edx-enterprise [Awais - wg-maintenance] [ 3.12 support PR ready to be merged]
- [x] acclaimbadge-xblock
- [x] ai-translations
- [x] help-tokens
- [x] enterprise-subsidy [IDA] [Umar]
- [x] edx-enterprise-subsidy-client [Umar]
- [x] openedx-ledger [Umar]
- [x] repo-tools [Umar]
- [x] event-bus-kafka [Usama] [Need to remove required checks]
- [x] api-manager [Usama] [Upgraded to 3.11]
- [x] auth-backends [Usama] [Internal review]
- [x] edx-drf-extensions [Usama]
- [x] edx-ora2 [Usama]
- [x] django-wiki [Usama]
- [x] edx-bulk-grades [Usama]
- [x] ccx-keys [Usama]
- [x] edx-enterprise-data [Usama]
- [x] edx-repo-health [Usama]
- [x] flashcards [Usama]
- [x] xblock-configuration
- [x] credentials-themes [Usama]
- [x] edx-django-utils [Usama]
- [x] edx-analytics-data-api-client [Usama]
- [x] edx-milestones [Usama]
- [x] edx-ace [Usama]
```
- [ ] Create a followup task to do the actual 3.11 upgrade and announce it to the org.

**Comment by jristau1984** [comment]
I believe we need a ticket spun off of this one to track getting devstack upgraded to 3.11.  @UsamaSadiq can you create this?

**Comment by UsamaSadiq** [comment]
Devstack was updated to have support for Python 3.11 in lms/cms images. Rest of the services have been updated to have Python 3.12 only the edx-analytics-dashboard repo is left which is in progress right now.

**Comment by MichaelRoytman** [comment]
@UsamaSadiq Hello! The following two repositories also need Python upgrades. Could you help us with this? Thank you!

* [learning-assistant](https://github.com/edx/learning-assistant)
* [persona-integration](https://github.com/edx/persona-integration)
